packages/controllers/src/cronjob/CronjobController.test.ts

```diff
@@ -0,0 +1,370 @@
+          return { [SnapEndowments.Cronjob]: MOCK_CRONJOB_PERMISSION } as any;
+        }
+        return false;
+      });
+
+    const cronjobController = new CronjobController({
+      messenger: controllerMessenger,
+    });
+
+    await cronjobController.register(MOCK_SNAP_ID);
+
+    expect(callActionMock).toHaveBeenCalledWith(
+      'PermissionController:getPermissions',
+      MOCK_SNAP_ID,
+    );
+
+    jest.advanceTimersByTime(inMilliseconds(1, Duration.Minute));
+
+    expect(callActionMock).toHaveBeenNthCalledWith(
+      4,
+      'SnapController:handleRequest',
+      {
+        snapId: MOCK_SNAP_ID,
+        origin: '',
+        handler: HandlerType.OnCronjob,
+        request: {
+          method: 'exampleMethodOne',
+          params: ['p1'],
+        },
+      },
+    );
+
+    cronjobController.destroy();
+  });
+
+  it('unregisters a cronjob', async () => {
+    const rootMessenger = getRootCronjobControllerMessenger();
+    const controllerMessenger =
+      getRestrictedCronjobControllerMessenger(rootMessenger);
+
+    const callActionMock = jest
+      .spyOn(controllerMessenger, 'call')
+      .mockImplementation((method) => {
+        if (method === 'SnapController:getAll') {
+          return [getTruncatedSnap()];
+        } else if (method === 'PermissionController:getPermissions') {
+          return { [SnapEndowments.Cronjob]: MOCK_CRONJOB_PERMISSION } as any;
+        }
+        return false;
+      });
+
+    const cronjobController = new CronjobController({
+      messenger: controllerMessenger,
+    });
+
+    await cronjobController.register(MOCK_SNAP_ID);
+    await cronjobController.unregister(MOCK_SNAP_ID);
+
+    jest.advanceTimersByTime(inMilliseconds(1, Duration.Minute));
+
+    expect(callActionMock).not.toHaveBeenNthCalledWith(
+      4,
+      'SnapController:handleRequest',
+      {
+        snapId: MOCK_SNAP_ID,
+        origin: '',
+        handler: HandlerType.OnCronjob,
+        request: {
+          method: 'exampleMethodOne',
+          params: ['p1'],
+        },
+      },
+    );
+
+    cronjobController.destroy();
+  });
+
+  it('executes cronjobs that were missed during daily check in', async () => {
+    const rootMessenger = getRootCronjobControllerMessenger();
+    const controllerMessenger =
+      getRestrictedCronjobControllerMessenger(rootMessenger);
+
+    const callActionMock = jest
+      .spyOn(controllerMessenger, 'call')
+      .mockImplementation((method) => {
+        if (method === 'SnapController:getAll') {
+          return [getTruncatedSnap()];
+        } else if (method === 'PermissionController:getPermissions') {
+          return { [SnapEndowments.Cronjob]: MOCK_CRONJOB_PERMISSION } as any;
+        }
+        return false;
+      });
+
+    const cronjobController = new CronjobController({
+      messenger: controllerMessenger,
+    });
+
+    // Update state manually for test
+    // eslint-disable-next-line dot-notation
+    cronjobController['update'](() => {
```

Comment:
```suggestion
    cronjobController.update(() => {
```

Reply:
I think that this one will have to stay like that only for that test case since the `update` method is `protected` method from `BaseController`.
```
TS2445: Property 'update' is protected and only accessible within class 'BaseController ' and its subclasses.
```

//cc @FrederikBolding

Reply:
You can use `ts-expect-error` for accessing private properties as well for tests - Not sure if that's something we want to do, but sometimes it is the only way around